### ISSUE TITLE
BUG: source .c files under _lib/src being gitignored.

### ISSUE THREAD

**Issue by dalejung** [pandas-dev/pandas#42116]
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of pandas.

- [x] (optional) I have confirmed this bug exists on the master branch of pandas.

Straight forward issue. Not all `.c` files are generated and shouldn't be ignored. Problematic for code searching tools that are set to respect `.gitignore` files. Ran into this when doing `rg` to search for `on_bad_lines` and never saw the matches in `_libs/src/tokenizer.c`

**Comment by simonjayhawkins** [comment]
Thanks @dalejung for the report. see also #41981 as somewhat related. Is there a way to keep parts of .gitignore and MANIFEST.in in sync

**Comment by lithomas1** [comment]
Nope, but the entries in gitignore for ``*.cpp`` and ``*.c`` can be straight up removed now, since Cython C/C++ files are not generated in the source tree anymore.

**Comment by jbrockmendel** [comment]
closed by #65474
